### PR TITLE
src: lib: dialog_ui: Add create form function

### PR DESCRIPTION
**Pull request by rodrigosiqueira**
Dialog provides support for creating simple forms. This commit a function to make it easy to create new forms.

### CODE REVIEW

**Comment by davidbtadokoro** [comment]
Hi @rodrigosiqueira, thanks for this change.
Going to look carefully at it as soon as possible.

File: src/lib/dialog_ui.sh [hunk 1]
```diff
@@ -625,6 +625,106 @@ function handle_exit()
...
+  # Change cancel label
+  cmd+=" --ok-label $'${ok_label}'"
+  cmd+=" --cancel-label $'${cancel_label}'"
+
+  # Add extra button
+  if [[ -n "$extra_label" ]]; then
+    cmd+=" --extra-button --extra-label ${extra_label}"
+  fi
+
+  # Creating form line
```
Comment (davidbtadokoro) [inline src/lib/dialog_ui.sh:709]: Maybe 'Adding form option' or 'Add form option'?

File: src/lib/dialog_ui.sh [hunk 2]
```diff
@@ -625,6 +625,106 @@ function handle_exit()
...
+    auxiliar_label_size=$(str_length "$field")
+    if [[ "${auxiliar_label_size}" -gt "${start_text_field}" ]]; then
+      start_text_field="${auxiliar_label_size}"
+    fi
+  done
+
+  # Add two extra spaces to the start point
+  start_text_field=$((start_text_field + 2))
+
+  # Change cancel label
```
Comment (davidbtadokoro) [inline src/lib/dialog_ui.sh:700]: Maybe 'Override OK and Cancel labels'? Either way, you are also changing the 'OK' label.

File: src/lib/dialog_ui.sh [hunk 3]
```diff
@@ -625,6 +625,106 @@ function handle_exit()
...
+
+  # Add layout to dialog
+  if [[ -n "$DIALOG_LAYOUT" ]]; then
+    cmd="DIALOGRC=${DIALOG_LAYOUT}"
+  fi
+
+  # Add general information to the dialog box
+  cmd+=" dialog --backtitle $'${back_title}' --title $'${box_title}' --clear --colors"
+
+  # Find out the largest label to know where we need to start the field
```
Comment (davidbtadokoro) [inline src/lib/dialog_ui.sh:689]: Maybe /need to start the field/need to start the **input** field/ ?

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
Thanks a lot, @rodrigosiqueira!
This screen surely will be handy all across the project. Made some comments, feel free to ping me with any doubts.

**Comment by rodrigosiqueira** [comment]
Hi @davidbtadokoro ,
I addressed all your comments. Thanks a lot for your review.

**Review by davidbtadokoro — APPROVED** [review]
Hi @rodrigosiqueira
Thanks for the update. It looks really nice to me 👍

**Pull request closed without merging**